### PR TITLE
Merge in Take Three at not Crashing from Roman

### PR DESCRIPTION
**Pull request by BarbourSmith**

### CODE REVIEW

File: FluidNC/src/Maslow/Maslow.cpp [hunk 1]
```diff
@@ -102,15 +118,15 @@ void printToWeb (double precision){
     log_info( "Calibration Precision: " << precision << "mm");
 }
 
-void Maslow_::readEncoders() {
-  axisTL.readEncoder();
-  axisTR.readEncoder();
-  axisBL.readEncoder();
-  axisBR.readEncoder();
-}
+// void Maslow_::readEncoders() {
```
Comment (BarbourSmith) [inline FluidNC/src/Maslow/Maslow.cpp:121]: Was this just not being called anywhere?

File: FluidNC/src/Protocol.cpp [hunk 2]
```diff
@@ -828,11 +828,17 @@ void protocol_exec_rt_system() {
...
-
-    Maslow.recomputePID(); //This one works as an alternative to having recomputePID called in DCServo.cpp
+  
+    //Maslow.recomputePID(); //This one works as an alternative to having recomputePID called in DCServo.cpp
 
     protocol_handle_events();
 
+    
+    //do all the Maslow stuff here
+    if(!Maslow.using_default_config && ( sys.state() == State::Jog || sys.state() == State::Cycle ) ){
```
Comment (BarbourSmith) [inline FluidNC/src/Protocol.cpp:838]: How does this work? Why do we only do it when sys.state == jog or cycle? Not complaining, just want to understand

**Review by BarbourSmith — COMMENTED** [review]
Looks good to me, and I tested that everything seems to be working well here 👍 👍

**Comment by BarbourSmith** [comment]
@azamat-bagatov  How do you feel about me merging this in? Does this look OK?

File: FluidNC/src/Protocol.cpp (hunk 2 above)
Comment (azamat-bagatov) [reply FluidNC/src/Protocol.cpp:838]: I _think_, those are the only two states where the FluidNC issues it commands to move. I was thinking about adding the Hold there too, but it seems to hold itself in place fine passively?

**Comment by azamat-bagatov** [comment]
Are you sure we're doing it the right way?)) 
image

File: FluidNC/src/Maslow/Maslow.cpp (hunk 1 above)
Comment (azamat-bagatov) [reply FluidNC/src/Maslow/Maslow.cpp:121]: yep

File: FluidNC/src/Protocol.cpp (hunk 2 above)
Comment (BarbourSmith) [reply FluidNC/src/Protocol.cpp:838]: Are we still updating the encoders the rest of the time? I'm pretty sure we need to do that all the time to make sure that they can't move when it's not looking

**Comment by azamat-bagatov** [comment]
Nice! Where can I see the changes you've merged?

**Comment by BarbourSmith** [comment]
If you click on the commit 

image

It will show you the changes in that commit like this:

https://github.com/BarbourSmith/FluidNC/pull/9/commits/51925df4f04b5bd2b336cd40640b7ef426828379

It looks like there were just some conflicts with white space that I needed to clean up

File: FluidNC/src/Protocol.cpp (hunk 2 above)
Comment (BarbourSmith) [reply FluidNC/src/Protocol.cpp:838]: Like we're probably not updating them?

**Pull request merged**